binary-decode.go

```diff
@@ -695,18 +696,158 @@ func (cdc *Codec) decodeReflectBinaryStruct(bz []byte, info *TypeInfo, rv reflec
 
 		// Read "StructTerm".
 		// NOTE: In the future, we'll need to break out of a loop
-		// when encoutering an StructTerm typ3 byte.
-		var typ = Typ3(0x00)
+		// when encountering an StructTerm typ3 byte.
+
+		var typ Typ3
```

Comment:
This and the comments above belong below the for-loop

Reply:
following 
> Then, line 716~723 below is unneeded

I deleted the part outside the for-loop and with it this comment(s).